### ISSUE TITLE
Possible Rate-Limiting issue, reported by Spacemacs users

### ISSUE THREAD

**Issue by tennysontbardwell** [melpa/melpa#8687]
MELPA maintainers,

Today[1], 10 people reported an issue ([link](https://github.com/syl20bnr/spacemacs/issues/16122)) in the Spacemacs repo.

The issue affects Spacemacs first-time startup across various versions of {OS, Emacs, Spacemacs} and is _suspected_ to be related to MELPA rate-limiting.

If you have any helpful thoughts, they would be highly appreciated.

[1]
In the last ~20hr

**Comment by milkypostman** [comment]
Yeah it's very likely we need to tweak the rate limiting. 

How many packages (roughly) does spacemacs need on first time initialization?

**Comment by milkypostman** [comment]
I will say that rate limiting has been active for over a year so it's not completely clear why this might just be noticed now.

**Comment by tennysontbardwell** [comment]
Odd. It seems to have stopped on its own (for me). I am not sure what investigatory steps might be helpful, if any. Feel free to close this issue at your discretion.

Number of packages:

* `spacemacs` install is 150 ([list](https://gist.github.com/tennysontbardwell/29e237bb6f00f8ce1649049200378499))
* `spacemacs-base` install is 60 lines ([list](https://gist.github.com/tennysontbardwell/147a2bf9d6fc50f1852bfcf1fa37c27f))
* My (approx) two dozen layers results in 400 packages ([list](https://gist.github.com/tennysontbardwell/f504f625387f7e7849a6150c17209975))

**Comment by purcell** [comment]
Hey, this was actually my fault, sorry — I messed things up for a day or two. I was experiencing rate-limiting in my own builds, specifically for `nix-emacs-ci`, which underpins the `setup-emacs` GitHub action. I tweaked the rate limiting, thinking I was just loosening it for the `archive-contents` manifest, but I misunderstood how the configuration directives work, and ended up _introducing_ limiting for package files where there previously was none. The limit was fairly high, but I realised what I'd done when I started hitting that too, figured out my mistake, and reversed it. Really sorry for any time wasted looking into this!
